zebra-state/src/service/finalized_state/tests/vectors.rs

```diff
@@ -85,9 +94,18 @@ fn sprout_note_commitment_tree_serialization_one() {
     let serialized_tree = incremental_tree.as_bytes();
     assert_eq!(hex::encode(&serialized_tree), expected_serialized_tree_hex);
 
-    let deserialized_tree = sprout::tree::NoteCommitmentTree::from_bytes(serialized_tree);
+    let deserialized_tree = sprout::tree::NoteCommitmentTree::from_bytes(&serialized_tree);
 
+    // This check isn't enough to show that the entire struct is the same, because it just compares
+    // the cached serialized/deserialized roots. (NoteCommitmentTree::eq() also just compares
+    // roots.)
     assert_eq!(incremental_tree.root(), deserialized_tree.root());
+
+    incremental_tree.assert_frontier_eq(&deserialized_tree);
```

Comment:
```suggestion
    incremental_tree.assert_frontier_eq(&deserialized_tree);

    // This check isn't enough to show that the entire struct is the same, because it just compares
    // the cached serialized/deserialized roots. (NoteCommitmentTree::eq() also just compares
    // roots.)
    assert_eq!(incremental_tree.root(), deserialized_tree.root());
```